tests/gradients/core/test_vjp.py

```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Tests for the gradients.vjp module."""
+# pylint: disable=import-outside-toplevel
```

Comment:
It's fine to leave it in, but for future reference you don't need this as `tests/.pylintrc` ignores this warning so any test files that are linted in CI won't raise this warning. You can lint locally with `pylint --rcfile tests/.pylintrc [path-to-test-file]`, which will give you a more accurate list of warnings for test files.

Reply:
Ohh, I always tried with `.pylintrc` (also the default I think) and got this warning, because it is not ignore in `.pylintrc` but just in `tests/.pylintrc`..

Reply:
Btw, why would we ignore `missing-class-docstring` or `missing-function-docstring`? We definitely want, at least test function, docstrings in test files, right?

Reply:
Not sure about why function doc strings are ignored, but I would assume class doc strings are ignored so that any dummy classes that are used within testing don't raise pylint warnings. Function docstrings could be ignored for the same reason perhaps. Although it doesn't make much sense to ignore those warnings in my opinion.

Reply:
Agreed. I also think that even dummy classes should come with a docstring. Adding them in this PR made me realize that some were duplicates of each other, leading to simplification :)

Reply:
It might be worth opening a PR that updates which pylint warnings are disabled in the `tests/.pylintrc` file.